### DISCUSSION TITLE
How does frigate handle full disk drives, and how to change it's storage space calculations?

### DISCUSSION THREAD

**Discussion by borgqueenx** [blakeblackshear/frigate#9308] [Ask A Question] [answered]
![image](https://github.com/blakeblackshear/frigate/assets/20416925/f73434c7-b5c4-4a4b-ad20-89084aa3502b)

The above image is my usage according to frigate, this is after using a fresh formatted drive, just added to network storage, and after deleting the /media/frigate folder on the internal drive, and after frigate started for 10 seconds the first time after doing all that.

I am using a hacky setup where i am using the samba nas add-on, to be able to connect my usb hard disk directly to the minipc running HAOS (not virtual). This is because HAOS does not easily allow you to mount a drive and use it as network drive for frigate to work on.
However it has a issue where frigate's files are on both the /media/frigate folder, as on the /frigate folder. 
When i calculate the total usage with windows going to that network drive, there are duplicated files of frigate media. However when also in windows, i delete one of those files, the one in the other folder gets deleted as well.

This is what windows shows me:
![image](https://github.com/blakeblackshear/frigate/assets/20416925/bafd46a8-5139-4102-9ac2-959e8e9e2cb4)

So I'm first of all wondering where this "starting" usage comes from, from the image i shared above, and secondly if there is a way for frigate to ignore a folder for calculating total disk space as it seems to be not a duplicate file, but something to do with mounting the drive.

Finally I am wondering what frigate will do if the disk space is full. and if it checks if it is really full, or if it uses that check for storage space used from the storage menu in the frigate UI for home assistant. Because those are two different things, clearly.

Lastly i want to provide some context on why:
- I had the hard drive connected to my router, this was working, however the speed was too low and frigate was skipping segments due to this. This issue is now solved with the drive directly connected to the HAOS minipc.
- Running HAOS as virtual OS has some other hacky things needed to make everything work, i am not good in the command line and therefore a system that is working out of the box is ideal for me. Furthermore, it would be a lot of work moving everything now from HAOS to something else, and I'm afraid i would break things in the process as well.
- I can't seem to find a guide on how to directly mount with HAOS to the /media folder so that i dont need network storage or samba.

Thanks for reading and for possibly providing assistance.

**Comment by NickM-27** [comment, marked as answer]
> if there is a way for frigate to ignore a folder for calculating total disk space as it seems to be not a duplicate file, but something to do with mounting the drive.

Like it shows on the tooltip, the storage page is an overview of total usage. Frigate uses the python library shutil for this, this is important because it does not cause drives to spin up which is important for users that run on HDD + SSD setup like Unraid or ZFS. This means that more specific calculations are not possible without spinning up drives.

> Finally I am wondering what frigate will do if the disk space is full. and if it checks if it is really full, or if it uses that check for storage space used from the storage menu in the frigate UI for home assistant. Because those are two different things, clearly.

When frigate calculates that there is less than 1 hour of space remaining for recording it will delete an hours worth of the oldest recording to ensure there is space to continue

**Reply by borgqueenx** [reply]
Thanks for the fast response Nick, so I can only use 2tb on this drive with the current setup, that's unfortunate. I'l take this to the HA forum, see if they can help me out further.
